src/modules/producer.js

```diff
@@ -173,13 +173,33 @@ class Producer {
         // reply to us if you receive this message!
         options.replyTo = this.amqpRPCQueues[queue].queue;
 
+        // convert timeout to amqp's expiration. It's message-level expiration.
+        // The message will be discarded from a queue once it’s been there longer than the given number of milliseconds
+        // This is needed to avoid the case when the message which is already expired from our pov (via timeout)
+        // is still waiting in the queue and thus is about to be processed by the consumer.
+        // Unfortunately, we can do nothing if the message is already consumed and is being processed at the moment
+        // when the timeout appears.
+        if (options.timeout && options.timeout > 0) {
+          if (this._connection.config.logger) {
+            // TODO: remove that warn after timeout option is removed in scope of major version update
+            this._connection.config.logger.warn({
```

Comment:
This will log warning every time the rpc is used. Maybe add a flag to log it only once?

Also, deprecation warnings are usually logged using `util.deprecate` which uses `process.emitWarning`, consider using it instead of the logger, like here: https://github.com/bringg/node-arnavmq/blob/master/src/index.js#L49

Reply:
thank you @ramhr . Done. I've never used this mechanism before. Thank you for pointing me to it.